### PR TITLE
[install] fix stale pointer with tarball URLs

### LINKED ISSUE #2512: Dependency `.tar.gz` support
### What version of Bun is running?

0.5.8

### What platform is your computer?

Darwin 22.4.0 arm64 arm

### What steps can reproduce the bug?

```
git clone https://github.com/nanoexpress/ultimate-oss.git --depth=1
cd ultimate-oss
bun install
```

### What is the expected behavior?

Successfully installation

### What do you see instead?

```sh
❯ bun install
bun install v0.5.8 (1a25af5e)
error: @nanoexpress/route-syntax-parser@https://github.com/nanoexpress/route-syntax-parser/releases/download/v2.4.4/nanoexpress-route-syntax-parser-2.4.4.tgz failed to resolve
error: uWebSockets.js@https://github.com/uNetworking/uWebSockets.js/archive/v20.20.0.tar.gz failed to resolve
ultimate on  master [✘!⇡] is 📦 v1.5.2 via ⬢ v19.8.1 took 29s 

```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by alexlamsl**
fixes #2512

### CODE REVIEW

**Comment by Jarred-Sumner** [comment]
At some point we will need to figure out how to free the URL strings

Probably will need to do reference counted strings (later)

File: src/install/install.zig [hunk 1]
```diff
@@ -2000,23 +2001,23 @@ pub const PackageManager = struct {
...
+            buf,
             "https://{s}/repos/{s}/{s}/tarball/{s}",
             .{
                 github_api_domain,
                 this.lockfile.str(&repository.owner),
                 this.lockfile.str(&repository.repo),
                 this.lockfile.str(&repository.committish),
             },
-        );
+        ) catch unreachable;
```
Comment (Jarred-Sumner) [inline src/install/install.zig:2020]: This isn't actually unreachable

It would be a buffer overflow

Comment (Jarred-Sumner) [reply src/install/install.zig:2020]: if combo of tag + repo + organization exceeded `1024` bytes (macOS) or `4096` bytes (Linux), then this buffer would overflow

Comment (alexlamsl) [reply src/install/install.zig:2020]: fixed in https://github.com/oven-sh/bun/pull/2520/commits/3736c55d43ef4e9340160b57f652a89bebb43ca9

**Comment by Jarred-Sumner** [comment]
Thank you

**Pull request merged**